test/unittests/CandidateTest.php

```diff
@@ -1174,7 +1174,7 @@ function testIsAccessibleByNoProject()
 
         $user = $this->getMockBuilder('\User')->getMock();
         $user->expects($this->atLeastOnce())->method("getCenterIDs")
-            ->willReturn([1, 2]);
+            ->willReturn([new \CenterID("1"), new \CenterID("2")]);
         $user->expects($this->atLeastOnce())->method("getProjectIDs")
             ->willReturn([new \ProjectID("2"), new \ProjectID("3")]);
         '@phan-var \User $user';
```

Comment:
```suggestion
  '@phan-var \User $user';   
$this->_dbMock->expects($this->once())
        ->method('pselectRow')
        ->willReturn("");
      
```

Reply:
The _setUpTestDoublesForSelectCandidate function defines the pselectRow only run once.
in the isAccessibleBy function, "pselectRow" will need to be called one more time.
So re-defines the method one more time will solve this issue.

Reply:
I don't understand why that would change here. All I've done is change the type from int to the new class, why would that affect the number of times that the function is called?

Reply:
@driusan I has been debugging this test case. $user is a mock class. It doesn't have the hasCenter method. 
It failed in here. 
 if ($projMatch && $centerMacth ) { 
     return true;
}
$centerMacth always returns null.
So it should add a method for hasCenter and return true. it should solve this test case.

Reply:
Thanks! That was the case since the check was previously inlined.. instead of mocking hasCenter to always returned true (which would default the purpose of the test..) I used `onlyMethods` so that hasCenter doesn't get replaced and it's passing the unit tests now.

Now to see if it passes the integration tests..